DIPs/1NNN-RAC.md

```diff
@@ -0,0 +1,333 @@
+
+## Contents
+* [Rationale](#rationale)
+* [Prior Work](#prior-work)
+* [Description](#description)
+* [Breaking Changes and Deprecations](#breaking-changes-and-deprecations)
+* [Reference](#reference)
+* [Glossary](#glossary)
+* [Copyright & License](#copyright--license)
+* [Reviews](#reviews)
+
+## Rationale
+
+Common use cases not being supported nor workaroundable can be quite frustrating to D users.
+Linker errors for those who have not studied linkers can be quite challenging to attempt an understanding of.
+With frequent request for help from those that reach them.
+
+A lot of problems with symbol representation can be silently hidden to D users when you only support static libraries and an executable for constructing a process containing D.
+However this is not the only configuration that people want or need their binaries to be built as.
+
+Plugins into D, D plugin into another language, replacable binaries are all examples of use cases for where someone may wish to use shared libraries that involve D.
+
+To make D work consistently and easily with a clear story for the user to understand, how symbols are represented in the language must be improved.
+
+The main consideration is to start at the module.
+If we know if the module is in the current binary, or outside of it, we can resolve a number of linker issues by default without any further modification.
+All other changes can come from this knowledge.
+
+## Prior Work
+
+In 2016 Benjamin Thaut did a talk at DConf ``D's Import and Export Business`` which covered a proposal for improving the language and the implementation details of export and shared libraries.
+
+This DIP, while not directly based upon it and does not touch upon the implementation details, does come to similar conclusions that ``export`` must not be a visibility modifier to make D work when using positive annotation.
+
+A key difference is that Benjamin's talk suggested that ``dllimport`` switch set to all was a good solution of specifying that a symbol is in DllImport mode.
+However after many years experience with D and shared libraries, it has been shown in the community usage that this is a source of constant linker errors and so a solution that is finer grained is required.
+
+## Description
+
+### Cornerstone Changes
+
+This section introduces crucial changes for future updates and ensures error-free compilation.
+
+1. External Import Path Switch: A new flag ``-extI`` is added, similar to the ``-I`` switch, marking modules as external to the currently compiling binary. This should be automated by build managers using the knowledge of dependencies being shared libraries to automatically apply it instead of ``-I``.
+2. Out-of-Binary Modules: For such modules, all non-templated scopes are interpreted to implicitly have ``extern`` attribute applied to them.
```

Comment:
What repurpose?

``export`` + ``extern`` already put a symbol into DllImport, I have stated that this is how you're expected to do it rather than just ``export`` by itself and that it is getting applied automatically to all scopes.